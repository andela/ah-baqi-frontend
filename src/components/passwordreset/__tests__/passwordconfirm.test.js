import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import rootReducer from '../../../reducers/index';
import ResetConfirm from '../passwordConfirmContainer';

const configureStoreItem = (initialState) => {
  const store = applyMiddleware(thunk)(createStore);
  return store(rootReducer, initialState);
};

describe('<ResetConfirm />', () => {
  const store = configureStoreItem();
  test('It renders correctly', () => {
    const wrapperComponent = mount(
      <Provider store={store}>
        <BrowserRouter>
          <ResetConfirm />
        </BrowserRouter>
      </Provider>,
    );
    const formItem = wrapperComponent.find(
      '[data-test="reset-form-cont"]',
    );
    expect(formItem).toHaveLength(2);

    const event = {
      target: { name: 'password', value: 'Whf34@jfFFdjkv34%#' },
      preventDefault: jest.fn(),
    };
    formItem.at(0).props().onSubmit(event);
    const pass = wrapperComponent.find(
      '[data-test="reset-form-cont-password"]',
    );
    pass.at(0).props().onChange(event);
    expect(pass).toHaveLength(2);

    const callback = jest.fn();
    pass.at(0).props()['data-__meta'].rules[1].validator.validator(
      'rule', 'Whf34@jfFFdjkv34%#', callback,
    );
    const confPass = wrapperComponent.find(
      '[data-test="reset-form-cont-reset-password"]',
    );
    confPass.at(0).props()['data-__meta'].rules[1].validator.validator(
      'rule', 'Whf34@jfFFdjkv34%#', callback,
    );
    wrapperComponent.setState({
      resetPassword: {
        password: 'Whf34@jfFFdjkv34%#',
      },
    });
    confPass.at(0).props()['data-__meta'].rules[1].validator.validator(
      'rule', 'Whf34@jfFFkv34%#', callback,
    );
    expect(wrapperComponent.state.resetPassword).toBe(undefined);
  });
});
