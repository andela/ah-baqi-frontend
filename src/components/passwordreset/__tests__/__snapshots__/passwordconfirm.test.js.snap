// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests PasswordResetConfirmContainer Test mounting of component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(Form(withRouter(ResetConfirm)))
      history={
        Object {
          "push": [MockFunction],
        }
      }
      match={
        Object {
          "params": Object {
            "token": "tcvygbuhijn6789",
          },
        }
      }
      showModal={[MockFunction]}
    >
      <Form(withRouter(ResetConfirm))
        history={
          Object {
            "push": [MockFunction],
          }
        }
        match={
          Object {
            "params": Object {
              "token": "tcvygbuhijn6789",
            },
          }
        }
        resetConfirmActions={[Function]}
        showModal={[MockFunction]}
      >
        <withRouter(ResetConfirm)
          form={
            Object {
              "getFieldDecorator": [Function],
              "getFieldError": [Function],
              "getFieldInstance": [Function],
              "getFieldProps": [Function],
              "getFieldValue": [Function],
              "getFieldsError": [Function],
              "getFieldsValue": [Function],
              "isFieldTouched": [Function],
              "isFieldValidating": [Function],
              "isFieldsTouched": [Function],
              "isFieldsValidating": [Function],
              "isSubmitting": [Function],
              "resetFields": [Function],
              "setFields": [Function],
              "setFieldsInitialValue": [Function],
              "setFieldsValue": [Function],
              "submit": [Function],
              "validateFields": [Function],
              "validateFieldsAndScroll": [Function],
            }
          }
          history={
            Object {
              "push": [MockFunction],
            }
          }
          match={
            Object {
              "params": Object {
                "token": "tcvygbuhijn6789",
              },
            }
          }
          resetConfirmActions={[Function]}
          showModal={[MockFunction]}
        >
          <ResetConfirm
            form={
              Object {
                "getFieldDecorator": [Function],
                "getFieldError": [Function],
                "getFieldInstance": [Function],
                "getFieldProps": [Function],
                "getFieldValue": [Function],
                "getFieldsError": [Function],
                "getFieldsValue": [Function],
                "isFieldTouched": [Function],
                "isFieldValidating": [Function],
                "isFieldsTouched": [Function],
                "isFieldsValidating": [Function],
                "isSubmitting": [Function],
                "resetFields": [Function],
                "setFields": [Function],
                "setFieldsInitialValue": [Function],
                "setFieldsValue": [Function],
                "submit": [Function],
                "validateFields": [Function],
                "validateFieldsAndScroll": [Function],
              }
            }
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            resetConfirmActions={[Function]}
            showModal={[MockFunction]}
          >
            <Form
              className="reset-form"
              colon={true}
              hideRequiredMark={false}
              layout="horizontal"
              onSubmit={[Function]}
              style={
                Object {
                  "textAlign": "left",
                }
              }
            >
              <form
                className="ant-form ant-form-horizontal reset-form"
                onSubmit={[Function]}
                style={
                  Object {
                    "textAlign": "left",
                  }
                }
              >
                <div>
                  <h1>
                    Create your new password
                  </h1>
                </div>
                <FormItem
                  data-test=""
                  hasFeedback={true}
                  label="Password"
                >
                  <Row
                    className="ant-form-item"
                    gutter={0}
                    key="row"
                  >
                    <div
                      className="ant-row ant-form-item"
                    >
                      <Col
                        className="ant-form-item-label"
                      >
                        <div
                          className="ant-col ant-form-item-label"
                        >
                          <label
                            className="ant-form-item-required"
                            htmlFor="password"
                            onClick={[Function]}
                            title="Password"
                          >
                            Password
                          </label>
                        </div>
                      </Col>
                      <Col
                        className="ant-form-item-control-wrapper"
                      >
                        <div
                          className="ant-col ant-form-item-control-wrapper"
                        >
                          <div
                            className="ant-form-item-control"
                          >
                            <span
                              className="ant-form-item-children"
                            >
                              <Input
                                data-__field={
                                  Object {
                                    "name": "password",
                                  }
                                }
                                data-__meta={
                                  Object {
                                    "name": "password",
                                    "originalProps": Object {
                                      "name": "password",
                                      "onChange": [Function],
                                      "placeholder": "*****",
                                      "prefix": <Icon
                                        style={
                                          Object {
                                            "color": "rgba(0,0,0,.25)",
                                          }
                                        }
                                        type="lock"
                                      />,
                                      "type": "password",
                                    },
                                    "ref": null,
                                    "rules": Array [
                                      Object {
                                        "message": "This field is required",
                                        "required": true,
                                      },
                                      Object {
                                        "validator": Object {
                                          "validator": [Function],
                                        },
                                      },
                                    ],
                                    "trigger": "onChange",
                                    "validate": Array [
                                      Object {
                                        "rules": Array [
                                          Object {
                                            "message": "This field is required",
                                            "required": true,
                                          },
                                          Object {
                                            "validator": Object {
                                              "validator": [Function],
                                            },
                                          },
                                        ],
                                        "trigger": Array [
                                          "onChange",
                                        ],
                                      },
                                    ],
                                    "valuePropName": "value",
                                  }
                                }
                                id="password"
                                name="password"
                                onChange={[Function]}
                                placeholder="*****"
                                prefix={
                                  <Icon
                                    style={
                                      Object {
                                        "color": "rgba(0,0,0,.25)",
                                      }
                                    }
                                    type="lock"
                                  />
                                }
                                type="password"
                              >
                                <span
                                  className="ant-input-affix-wrapper"
                                >
                                  <span
                                    className="ant-input-prefix"
                                  >
                                    <Icon
                                      style={
                                        Object {
                                          "color": "rgba(0,0,0,.25)",
                                        }
                                      }
                                      type="lock"
                                    >
                                      <LocaleReceiver
                                        componentName="Icon"
                                      >
                                        <i
                                          aria-label="icon: lock"
                                          className="anticon anticon-lock"
                                          style={
                                            Object {
                                              "color": "rgba(0,0,0,.25)",
                                            }
                                          }
                                        >
                                          <IconReact
                                            className=""
                                            type="lock-o"
                                          >
                                            <svg
                                              aria-hidden="true"
                                              className=""
                                              data-icon="lock"
                                              fill="currentColor"
                                              focusable="false"
                                              height="1em"
                                              key="svg-lock"
                                              viewBox="64 64 896 896"
                                              width="1em"
                                            >
                                              <path
                                                d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0z"
                                                key="svg-lock-svg-0"
                                              />
                                            </svg>
                                          </IconReact>
                                        </i>
                                      </LocaleReceiver>
                                    </Icon>
                                  </span>
                                  <input
                                    className="ant-input"
                                    data-__field={
                                      Object {
                                        "name": "password",
                                      }
                                    }
                                    data-__meta={
                                      Object {
                                        "name": "password",
                                        "originalProps": Object {
                                          "name": "password",
                                          "onChange": [Function],
                                          "placeholder": "*****",
                                          "prefix": <Icon
                                            style={
                                              Object {
                                                "color": "rgba(0,0,0,.25)",
                                              }
                                            }
                                            type="lock"
                                          />,
                                          "type": "password",
                                        },
                                        "ref": null,
                                        "rules": Array [
                                          Object {
                                            "message": "This field is required",
                                            "required": true,
                                          },
                                          Object {
                                            "validator": Object {
                                              "validator": [Function],
                                            },
                                          },
                                        ],
                                        "trigger": "onChange",
                                        "validate": Array [
                                          Object {
                                            "rules": Array [
                                              Object {
                                                "message": "This field is required",
                                                "required": true,
                                              },
                                              Object {
                                                "validator": Object {
                                                  "validator": [Function],
                                                },
                                              },
                                            ],
                                            "trigger": Array [
                                              "onChange",
                                            ],
                                          },
                                        ],
                                        "valuePropName": "value",
                                      }
                                    }
                                    id="password"
                                    name="password"
                                    onChange={[Function]}
                                    onKeyDown={[Function]}
                                    placeholder="*****"
                                    style={null}
                                    type="password"
                                    value=""
                                  />
                                </span>
                              </Input>
                            </span>
                            <Animate
                              animation={Object {}}
                              component=""
                              componentProps={Object {}}
                              key="help"
                              onAppear={[Function]}
                              onEnd={[Function]}
                              onEnter={[Function]}
                              onLeave={[Function]}
                              transitionAppear={true}
                              transitionEnter={true}
                              transitionLeave={true}
                              transitionName="show-help"
                            />
                          </div>
                        </div>
                      </Col>
                    </div>
                  </Row>
                </FormItem>
                <FormItem
                  data-test=""
                  hasFeedback={true}
                  label="Confirm Password"
                >
                  <Row
                    className="ant-form-item"
                    gutter={0}
                    key="row"
                  >
                    <div
                      className="ant-row ant-form-item"
                    >
                      <Col
                        className="ant-form-item-label"
                      >
                        <div
                          className="ant-col ant-form-item-label"
                        >
                          <label
                            className="ant-form-item-required"
                            htmlFor="confirm_password"
                            onClick={[Function]}
                            title="Confirm Password"
                          >
                            Confirm Password
                          </label>
                        </div>
                      </Col>
                      <Col
                        className="ant-form-item-control-wrapper"
                      >
                        <div
                          className="ant-col ant-form-item-control-wrapper"
                        >
                          <div
                            className="ant-form-item-control"
                          >
                            <span
                              className="ant-form-item-children"
                            >
                              <Input
                                data-__field={
                                  Object {
                                    "name": "confirm_password",
                                  }
                                }
                                data-__meta={
                                  Object {
                                    "name": "confirm_password",
                                    "originalProps": Object {
                                      "name": "confirm_password",
                                      "onChange": [Function],
                                      "placeholder": "*****",
                                      "prefix": <Icon
                                        style={
                                          Object {
                                            "color": "rgba(0,0,0,.25)",
                                          }
                                        }
                                        type="lock"
                                      />,
                                      "type": "password",
                                    },
                                    "ref": null,
                                    "rules": Array [
                                      Object {
                                        "message": "This field is required",
                                        "required": true,
                                      },
                                      Object {
                                        "validator": Object {
                                          "validator": [Function],
                                        },
                                      },
                                    ],
                                    "trigger": "onChange",
                                    "validate": Array [
                                      Object {
                                        "rules": Array [
                                          Object {
                                            "message": "This field is required",
                                            "required": true,
                                          },
                                          Object {
                                            "validator": Object {
                                              "validator": [Function],
                                            },
                                          },
                                        ],
                                        "trigger": Array [
                                          "onChange",
                                        ],
                                      },
                                    ],
                                    "valuePropName": "value",
                                  }
                                }
                                id="confirm_password"
                                name="confirm_password"
                                onChange={[Function]}
                                placeholder="*****"
                                prefix={
                                  <Icon
                                    style={
                                      Object {
                                        "color": "rgba(0,0,0,.25)",
                                      }
                                    }
                                    type="lock"
                                  />
                                }
                                type="password"
                              >
                                <span
                                  className="ant-input-affix-wrapper"
                                >
                                  <span
                                    className="ant-input-prefix"
                                  >
                                    <Icon
                                      style={
                                        Object {
                                          "color": "rgba(0,0,0,.25)",
                                        }
                                      }
                                      type="lock"
                                    >
                                      <LocaleReceiver
                                        componentName="Icon"
                                      >
                                        <i
                                          aria-label="icon: lock"
                                          className="anticon anticon-lock"
                                          style={
                                            Object {
                                              "color": "rgba(0,0,0,.25)",
                                            }
                                          }
                                        >
                                          <IconReact
                                            className=""
                                            type="lock-o"
                                          >
                                            <svg
                                              aria-hidden="true"
                                              className=""
                                              data-icon="lock"
                                              fill="currentColor"
                                              focusable="false"
                                              height="1em"
                                              key="svg-lock"
                                              viewBox="64 64 896 896"
                                              width="1em"
                                            >
                                              <path
                                                d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0z"
                                                key="svg-lock-svg-0"
                                              />
                                            </svg>
                                          </IconReact>
                                        </i>
                                      </LocaleReceiver>
                                    </Icon>
                                  </span>
                                  <input
                                    className="ant-input"
                                    data-__field={
                                      Object {
                                        "name": "confirm_password",
                                      }
                                    }
                                    data-__meta={
                                      Object {
                                        "name": "confirm_password",
                                        "originalProps": Object {
                                          "name": "confirm_password",
                                          "onChange": [Function],
                                          "placeholder": "*****",
                                          "prefix": <Icon
                                            style={
                                              Object {
                                                "color": "rgba(0,0,0,.25)",
                                              }
                                            }
                                            type="lock"
                                          />,
                                          "type": "password",
                                        },
                                        "ref": null,
                                        "rules": Array [
                                          Object {
                                            "message": "This field is required",
                                            "required": true,
                                          },
                                          Object {
                                            "validator": Object {
                                              "validator": [Function],
                                            },
                                          },
                                        ],
                                        "trigger": "onChange",
                                        "validate": Array [
                                          Object {
                                            "rules": Array [
                                              Object {
                                                "message": "This field is required",
                                                "required": true,
                                              },
                                              Object {
                                                "validator": Object {
                                                  "validator": [Function],
                                                },
                                              },
                                            ],
                                            "trigger": Array [
                                              "onChange",
                                            ],
                                          },
                                        ],
                                        "valuePropName": "value",
                                      }
                                    }
                                    id="confirm_password"
                                    name="confirm_password"
                                    onChange={[Function]}
                                    onKeyDown={[Function]}
                                    placeholder="*****"
                                    style={null}
                                    type="password"
                                    value=""
                                  />
                                </span>
                              </Input>
                            </span>
                            <Animate
                              animation={Object {}}
                              component=""
                              componentProps={Object {}}
                              key="help"
                              onAppear={[Function]}
                              onEnd={[Function]}
                              onEnter={[Function]}
                              onLeave={[Function]}
                              transitionAppear={true}
                              transitionEnter={true}
                              transitionLeave={true}
                              transitionName="show-help"
                            />
                          </div>
                        </div>
                      </Col>
                    </div>
                  </Row>
                </FormItem>
                <FormItem
                  hasFeedback={false}
                >
                  <Row
                    className="ant-form-item"
                    gutter={0}
                    key="row"
                  >
                    <div
                      className="ant-row ant-form-item"
                    >
                      <Col
                        className="ant-form-item-control-wrapper"
                      >
                        <div
                          className="ant-col ant-form-item-control-wrapper"
                        >
                          <div
                            className="ant-form-item-control"
                          >
                            <span
                              className="ant-form-item-children"
                            >
                              <Button
                                block={true}
                                className="login-form-button"
                                ghost={false}
                                htmlType="submit"
                                loading={false}
                                type="primary"
                              >
                                <Wave>
                                  <button
                                    className="ant-btn login-form-button ant-btn-primary ant-btn-block"
                                    onClick={[Function]}
                                    type="submit"
                                  >
                                    <span
                                      key=".0"
                                    >
                                      Reset Password
                                    </span>
                                  </button>
                                </Wave>
                              </Button>
                            </span>
                            <Animate
                              animation={Object {}}
                              component=""
                              componentProps={Object {}}
                              key="help"
                              onAppear={[Function]}
                              onEnd={[Function]}
                              onEnter={[Function]}
                              onLeave={[Function]}
                              transitionAppear={true}
                              transitionEnter={true}
                              transitionLeave={true}
                              transitionName="show-help"
                            />
                          </div>
                        </div>
                      </Col>
                    </div>
                  </Row>
                </FormItem>
              </form>
            </Form>
          </ResetConfirm>
        </withRouter(ResetConfirm)>
      </Form(withRouter(ResetConfirm))>
    </Connect(Form(withRouter(ResetConfirm)))>
  </Provider>
</Router>
`;
